### PR TITLE
Issue 957. Rectified the resize Logic.

### LINKED ISSUE #957: Max Worker - Worker Change - Edit - Resize - Data Not Capturing - Issue
In the event of changing the cluster size from the configuration while the cluster is running, it is considered a resize and not an edit to the cluster.
This is currently not getting captured in Overwatch.
We need to understand different scenarios on the data and work on it in the code.

### PR DESCRIPTION
**Pull request by brij-raghuwanshi-db**
closes #957 issue

### CODE REVIEW

**Comment by gueniai** [comment]
hey @brij-raghuwanshi-db , this is going to main, can you please rebase?

**Comment by brij-raghuwanshi-db** [comment]
Done, Please verify.


*Brijendra Singh Raghuwanshi*
*SSA*
+91-8551031355
www.databricks.com

**Review by souravbaner-da — APPROVED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala [hunk 1]
```diff
@@ -721,7 +721,7 @@ trait SilverTransforms extends SparkSessionWrapper {
     val clusterBaseDF = clusterBase(df)
     val clusterBaseWMetaDF = clusterBaseDF
       // remove start, startResults, and permanentDelete as they do not contain sufficient metadata
-      .filter('actionName.isin("create", "edit"))
+      .filter('actionName.isin("create", "edit", "resize"))
```
Comment (gueniai) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:724]: Brij I reviewed the PR Proof and in the Include Resize notebook, command 38 I was expecting to see some counts for resize in the "after" section. What am I missing?

**Review by gueniai — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala (hunk 1 above)
Comment (brij-raghuwanshi-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:724]: For this activity, the main idea was to ensure that the schema does not change in the event of occurence of RESIZE relevant data after inclusion in the audit log bronze.

Since our environment was not having the data relevant to RESIZE, we have the supporting dataset from the another environment. Which is why, the schema comparison was done another notebook that was linked in the original support notebook.

To summarise:
**PR_957_Include_Resize** does not have required datapoint to compare the schema; so, **PR_957_Check_Data** is executed in the another environment and the schema comparison was done there. The notebook was then exported from that environment and put here.

Please let me know if this helps.

Comment (gueniai) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:724]: @brij-raghuwanshi-db  thank you! That does help. I just checked that other notebook, and I see that we get the events in Audit logs, but I don't see them in our table. Can you run your code on that other workspace to make sure these events are in fact flowing into the table?

Comment (brij-raghuwanshi-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:724]: The Other Environment is a customer Environment. Let me know your thoughts.

**Review by gueniai — APPROVED** [review]
